bundles/com.espressif.idf.launch.serial.core/src/com/espressif/idf/launch/serial/internal/SerialFlashLaunchConfigDelegate.java

```diff
@@ -88,6 +89,16 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
 		// Start the launch (pause the serial port)
 		((SerialFlashLaunch) launch).start();
 		boolean isFlashOverJtag = configuration.getAttribute(IDFLaunchConstants.FLASH_OVER_JTAG, false);
+		serialPort = ((SerialFlashLaunch) launch).getLaunchTarget()
+				.getAttribute(SerialFlashLaunchTargetProvider.ATTR_SERIAL_PORT, ""); //$NON-NLS-1$
+
+		if (DfuCommandsUtil.isDfu()) {
+			if (checkIfPortIsEmpty(configuration)) {
+				return;
+			}
+			DfuCommandsUtil.flashDfuBins(getProject(configuration), launch, monitor, serialPort);
+			return;
+		}
```

Comment:
Code not formatted properly here as well

Reply:
fixed